user/timepix3/module/src/Timepix3Event2StdEventConverter.cc

```diff
@@ -56,15 +68,23 @@ bool Timepix3TrigEvent2StdEventConverter::Converting(eudaq::EventSPC ev, eudaq::
   timestamp = timestamp_raw + (static_cast<long long int>(m_TDCoverflowCounter) << 35);
 
   // Calculate timestamp in picoseconds assuming 320 MHz clock:
-  uint64_t triggerTime = (timestamp + static_cast<long long int>(stamp) / 12) * 3125;
+  uint64_t triggerTime = static_cast<uint64_t>((static_cast<double>(timestamp) + static_cast<double>(stamp) / 12.) * 3125);
```

Comment:
Done as in Corryvreckan 3907485db3ee79ec0b92bd778988a5f3e650cd77.

The only difference is that we need another `static_cast<uint64_t>` because here `triggerTime` is a `uint64_t` in units of picoseconds whereas we're using a double in units of nanoseconds in Corryvreckan.

Reply:
Sorry, but then this doesn't make sense - you're casting to a double for a division and then cast back to an integer? Please do your math properly.

Reply:
What about 
```cpp
uint64_t triggerTime = timestamp*3125 + static_cast<uint64_t>(stamp * (3125/12.));
```

I'm aware of the fact that I'm introducing some rounding errors here but `stamp` needs to be added to the coarse timestamp and is given as a fraction of <0-11>/12, and 3125/12 is not an integer unfortunately.
But due to the picosecond binning of `triggerTime` we cannot express this timestamp more precisely.

Reply:
You decrease your rounding error if you first multiply `stamp` with the conversion factor and then divide.

Reply:
So then we should go for:
```cpp
static_cast<uint64_t>((stamp * 3125) / 12.);
```

Reply:
no, that's still a double division. don't overcomplicate things... :)

Reply:
Okay, I think I got it now:
```cpp
(stamp * 3125) / 12;
```